core/model/modx/modcachemanager.class.php

```diff
@@ -756,4 +756,22 @@ public function clearCache(array $paths= array(), array $options= array()) {
 
         return $results;
     }
+
+    /**
+     * Flush permissions for users
+     *
+     * @return string|array The success response
+     */
+    public function flushPermissions() {
+        $ctxQuery = $this->modx->newQuery('modContext');
+        $ctxQuery->select($this->modx->getSelectColumns('modContext', '', '', array('key')));
+        if ($ctxQuery->prepare() && $ctxQuery->stmt->execute()) {
+            $contexts = $ctxQuery->stmt->fetchAll(PDO::FETCH_COLUMN);
+            if ($contexts) {
+                $serialized = serialize($contexts);
```

Comment:
After some testing, turns out this should not be required, xPDO should handle the serialization on its own.